sdk/data_cosmos/src/operations/patch_document.rs

```diff
@@ -0,0 +1,186 @@
+use crate::headers::from_headers::*;
+use crate::prelude::*;
+use crate::resources::document::DocumentAttributes;
+use crate::ResourceQuota;
+
+use azure_core::headers::session_token_from_headers;
+use azure_core::Response as HttpResponse;
+use azure_core::SessionToken;
+use serde::Serialize;
+use serde_json::Value;
+use time::OffsetDateTime;
+
+operation! {
+    PatchDocument,
+    client: DocumentClient,
+    operations: Vec<Operation>,
+    ?condition: String
+}
+
+impl PatchDocumentBuilder {
+    pub fn into_future(self) -> PatchDocument {
+        Box::pin(async move {
+            let mut request = self.client.document_request(azure_core::Method::Patch);
+
+            crate::cosmos_entity::add_as_partition_key_header_serialized(
+                self.client.partition_key_serialized(),
+                &mut request,
+            );
+
+            let patch_request = PatchDocumentRequest {
+                condition: self.condition,
+                operations: self.operations,
+            };
+
+            let serialized = azure_core::to_json(&patch_request)?;
+            request.set_body(serialized);
+
+            let response = self
+                .client
+                .cosmos_client()
+                .pipeline()
+                .send(
+                    self.context.clone().insert(ResourceType::Documents),
+                    &mut request,
+                )
+                .await?;
+
+            PatchDocumentResponse::try_from(response).await
+        })
+    }
+}
+
+#[derive(Serialize, Debug)]
+struct PatchDocumentRequest {
+    #[serde(skip_serializing_if = "Option::is_none")]
+    condition: Option<String>,
+    operations: Vec<Operation>,
+}
+
+#[derive(Serialize, Debug, Clone)]
+#[serde(rename_all = "lowercase")]
+#[serde(tag = "op")]
+pub enum Operation {
+    Add { path: String, value: Value },
+    Remove { path: String },
+    Set { path: String, value: Value },
+    Incr { path: String, value: Value },
```

Comment:
Given the documentation specifies this must be a positive or negative value, what are your thoughts on having this be a `serde_json::value::Number`?

Reply:
That makes sense to me.